src/client/dfuse/ops/readdir.c

```diff
@@ -147,36 +147,36 @@ dfuse_cb_readdir(fuse_req_t req, struct dfuse_inode_entry *inode,
 			rc = dfs_lookup_rel(inode->ie_dfs->dffs_dfs,
 					    inode->ie_obj, dirents[i].d_name,
 					    O_RDONLY, &ie->ie_obj, &mode);
-			if (rc != -DER_SUCCESS) {
+			if (rc) {
 				D_FREE(ie);
-				D_GOTO(err_or_buf, 0);
+				D_GOTO(err_or_buf, rc = -rc);
 			}
 
 			rc = dfs_ostat(inode->ie_dfs->dffs_dfs, ie->ie_obj,
 				       &ie->ie_stat);
-			if (rc != -DER_SUCCESS) {
+			if (rc) {
```

Comment:
At line 137 in src/client/dfuse/dfuse_cont.c, we check against DER_SUCCESS, we should probably try to be consistent here.